### PR TITLE
ransack with js for sform search chat rooms

### PR DESCRIPTION
**Pull request by nguyenngocha**

### CODE REVIEW

File: app/assets/javascripts/chat.js [hunk 1]
```diff
@@ -8,6 +8,13 @@ $(document).on("turbolinks:load", function() {
     height: "86%",
     size: "6px"
   });
+
+  $('.search-chat-room').on('keyup', function() {
+    if (this.value.length >= 1) {
```
Comment (NgengChhengkim) [inline app/assets/javascripts/chat.js]: this.value.trim() !== ''

File: app/controllers/chats_controller.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
 class ChatsController < ApplicationController
   def index
     @courses = current_user.courses
-    @users = User.per_page_kaminari(params[:page]).per Settings.chats.chat_room_per_page
+    @users = User.all
```
Comment (NgengChhengkim) [inline app/controllers/chats_controller.rb]: bo cai nay

File: app/controllers/chats_controller.rb [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+
+    @users = @users.ransack(name_cont: params[:q]).result if params[:q].present?
```
Comment (NgengChhengkim) [inline app/controllers/chats_controller.rb]: thu cai nay: User.ransack

File: app/controllers/chats_controller.rb [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
+    @users = @users.per_page_kaminari(params[:page]).per Settings.per_page
+    @courses = @courses.ransack(name_cont: params[:q]).result if params[:q].present?
```
Comment (NgengChhengkim) [inline app/controllers/chats_controller.rb]: bo if

File: app/assets/javascripts/chat.js [hunk 5]
```diff
@@ -8,6 +8,13 @@ $(document).on("turbolinks:load", function() {
     height: "86%",
     size: "6px"
   });
+
+  $('.search-chat-room').on('keyup', function() {
+    if (this.value.trim() !== '') {
+      $(".search-chats").data("search", "1");
```
Comment (NgengChhengkim) [inline app/assets/javascripts/chat.js]: dung `''`

File: app/views/shared/_chat_rooms.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  <%= render partial: "shared/chat_room", collection: @users, as: :chat_room %>
+</div>
+<div id="infinite-scrolling">
+  <div class="loading" hidden>
+    <i class="fa fa-spinner fa-pulse fa-2x fa-fw"></i>
+    <span class="sr-only"><%= t "chats.loading" %></span>
+  </div>
+  <div hidden><%= paginate @users %></div>
+</div>
+<div class="search-chats" data-search="0">
```
Comment (NgengChhengkim) [inline app/views/shared/_chat_rooms.html.erb]: dat ten data khac di de de hieu. vd data-type="search"

File: app/assets/javascripts/chat.js [hunk 7]
```diff
@@ -8,16 +8,24 @@ $(document).on("turbolinks:load", function() {
     height: "86%",
     size: "6px"
   });
+
+  $(".search-chat-room").on("keyup", function() {
+    if (this.value.trim() !== "") {
+      $("#users").html("");
+      paginate();
```
Comment (yongsokheng) [inline app/assets/javascripts/chat.js]: dat ten nao khac xem: vidu: load_more_chat_rooms(), ..

File: app/assets/javascripts/pagination.js [hunk 8]
```diff
@@ -1,13 +1,17 @@
...
+  paginate();
+});
+
+function paginate() {
+  if ($("#infinite-scrolling").size() > 0) {
+    $("#chat-sidebar").bindWithDelay("scroll", function() {
       var more_posts_url;
-      more_posts_url = $('.pagination .next a').attr('href');
+      more_posts_url = $(".pagination .next a").attr("href");
       if (more_posts_url && $(window).scrollTop() > $(document).height() - $(window).height() - 60) {
```
Comment (yongsokheng) [inline app/assets/javascripts/pagination.js]: doi $(window).scrollTop() thanh $(this).scrollTop()

File: app/assets/stylesheets/chat_layout.scss [hunk 9]
```diff
@@ -80,6 +80,16 @@
...
     }
+    .center {
+      text-align: center; 
+    }
+    .search-chat-room {
+      padding-left: 25px;
+      position: fixed;
+    }
+    #users{
+      padding-top:31px;
```
Comment (yongsokheng) [inline app/assets/stylesheets/chat_layout.scss]: convention

File: app/assets/stylesheets/chat_layout.scss [hunk 10]
```diff
@@ -80,6 +80,16 @@
...
       }
     }
+    .center {
+      text-align: center; 
+    }
+    .search-chat-room {
+      padding-left: 25px;
+      position: fixed;
+    }
+    #users{
```
Comment (yongsokheng) [inline app/assets/stylesheets/chat_layout.scss]: cach mot space

File: app/controllers/chats_controller.rb [hunk 11]
```diff
@@ -1,7 +1,8 @@
 class ChatsController < ApplicationController
   def index
-    @courses = current_user.courses
-    @users = User.per_page_kaminari(params[:page]).per Settings.chats.chat_room_per_page
+    @users = User.ransack(name_cont: params[:q]).result
+      .per_page_kaminari(params[:page]).per 10
```
Comment (yongsokheng) [inline app/controllers/chats_controller.rb]: dung Settings

**Pull request merged**